app/services/purl_and_stacks_service.rb

```diff
@@ -56,6 +60,16 @@ def delete(version:)
     UpdatePurlMetadataService.new(purl).delete! if legacy_purl_enabled?
   end
 
+  # Withdraw or restore a version.
+  # @param version [String] the version number
+  # @param withdrawn [Boolean] true to withdraw, false to restore
+  def withdraw(version:, withdrawn: true)
+    # unversioned is an implicit version 1, which cannot be withdrawn.
+    raise VersionedFilesService::BadRequestError, 'Cannot withdraw head version' unless already_versioned_layout?
+
+    versioned_files_service.withdraw(version:, withdrawn:)
+  end
```

Comment:
I guess the logic hits the versioned files service via this service because this is the class responsible for determining if the layout is versioned?

Reply:
PurlAndStacksService has been serving as the level of abstraction that is aware of version layout.

Reply:
👍🏻 Makes sense to isolate that so it's not leaked all over.